fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -926,4 +960,40 @@ private String getAccessToken() throws FHIRPersistenceInterceptorException {
                 .filter(s -> !s.isEmpty())
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Gets the resource type of the resource represented in the bundle entry.
+     * @param entry the bundle entry
+     * @return the resource type, or null
+     */
+    private String getResourceType(Bundle.Entry entry) {
+        Resource resource = entry.getResource();
+        if (resource != null) {
+            return resource.getClass().getSimpleName();
+        } else if (entry.getFullUrl() != null && entry.getFullUrl().getValue() != null) {
+            String[] urlParts = entry.getFullUrl().getValue().split("/");
```

Comment:
this presumes that we'll never start using absolute URLs in our fullUrls, which I think it a bit suspect.
could we make it resilient to that?

Reply:
if not, maybe just update the javadoc to be very clear about this assumption?

Reply:
Doesn't this code work for both literal and absolute URLs?  Or am I missing something?

Reply:
i think I just misread it as `urlParts[2]` ... we never echo parameters back in the fullUrl and we no longer have fullUrl that populate `[type]/[id]/_history/[vid]`, so I agree this should work